src/Native/System.Native/pal_networking.cpp

```diff
@@ -0,0 +1,332 @@
+    {
+        sockaddr_in* addr = reinterpret_cast<sockaddr_in*>(info.ai_addr);
+        ConvertSockAddrInToByteArray(buffer, bufferLength, *addr);
+        *port = addr->sin_port;
+    });
+
+    return ConvertGetAddrInfoAndGetNameInfoErrorsToPal(result);
+}
+
+extern "C" int32_t IpAddressToString(const char* address, int32_t addressLength, bool isIpv6, char* string, int32_t stringLength, uint32_t scope/* = 0*/)
+{
+    assert(address != nullptr);
+    assert((addressLength == NUM_BYTES_IN_IPV6_ADDRESS) || (addressLength == NUM_BYTES_IN_IPV4_ADDRESS));
+    assert(string != nullptr);
+
+    // These constants differ per platform so the managed side uses the bigger value; therefore, check that
+    // the length is between the two lengths
+    assert((stringLength >= INET_ADDRSTRLEN) && (stringLength <= INET6_ADDRSTRLEN));
+    (void) addressLength; // We only need this to be sure the address is the right length
+
+    int32_t result;
+    socklen_t len = UnsignedCast(stringLength);
+
+    if (isIpv6)
+    {
+        sockaddr_in6 addr = {};
+        addr.sin6_scope_id = scope;
+        ConvertByteArrayToV6SockAddrIn(addr, address, addressLength);
+        result = getnameinfo(reinterpret_cast<const sockaddr*>(&addr), sizeof(sockaddr_in6), string, len, nullptr, 0, NI_NUMERICHOST);
+    }
+    else
+    {
+        sockaddr_in addr = {};
+        ConvertByteArrayToSockAddrIn(addr, address, addressLength);
+        result = getnameinfo(reinterpret_cast<const sockaddr*>(&addr), sizeof(sockaddr_in), string, len, nullptr, 0, NI_NUMERICHOST);
+    }
+
+    return ConvertGetAddrInfoAndGetNameInfoErrorsToPal(result);
+}
+
+extern "C" int32_t GetHostEntriesForName(const char* address, HostEntry** entry)
+{
+    assert(address != nullptr);
+    assert(entry != nullptr);
+
+    // Get all address families and the canonical name
+    addrinfo hint = {};
+    hint.ai_family = AF_UNSPEC;
+    hint.ai_flags = AI_CANONNAME;
+
+    addrinfo* info = nullptr;
+    int result = getaddrinfo(address, nullptr, &hint, &info);
+    if (result != 0)
+    {
+        return ConvertGetAddrInfoAndGetNameInfoErrorsToPal(result);
+    }
+
+    // Allocate our data and ensure everything is 0
+    *entry = new HostEntry;
+    HostEntry* he = *entry;
+    memset(he, 0, sizeof(HostEntry));
+
+    std::vector<addrinfo*> infos;
+
+    // Walk the linked-list of entries and get the IP Addresses
+    for (addrinfo* ai = info; ai != nullptr; ai = ai->ai_next)
+    {
+        // If we haven't found a canonical name yet and this addrinfo has one, copy it
+        if ((he->CanonicalName == nullptr) && (ai->ai_canonname != nullptr))
+        {
+            size_t len = strlen(ai->ai_canonname) + 1; // Name + null
+            he->CanonicalName = new char[len];
+            SafeStringCopy(he->CanonicalName, len, ai->ai_canonname);
+        }
+
+        // Skip non-IPv4 and non-IPv6 addresses
+        if ((ai->ai_family != AF_INET) && (ai->ai_family != AF_INET6))
+            continue;
```

Comment:
(there are a few other places in this file that have the same issue, e.g. line 291: https://github.com/dotnet/corefx/pull/3471/files#diff-a533f7324e00c23d36788e9084520906R291)